tony-history-server/app/controllers/HomeController.java

```diff
@@ -0,0 +1,102 @@
+package controllers;
+
+import com.typesafe.config.Config;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import javax.inject.Inject;
+
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.HdfsConfiguration;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.yarn.api.ApplicationConstants;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+
+import play.Logger.ALogger;
+import play.Logger;
+import play.mvc.*;
+
+
+public class HomeController extends Controller {
+  private static final ALogger LOG = Logger.of(HomeController.class);
+  private final Config config;
+
+  private final String HADOOP_CONF_DIR = ApplicationConstants.Environment.HADOOP_CONF_DIR.key();
+  private final String CORE_SITE_CONF = YarnConfiguration.CORE_SITE_CONFIGURATION_FILE;
+  private final String HDFS_SITE_CONF = "hdfs-site.xml";
+
+  @Inject
+  public HomeController(Config config) {
+    this.config = config;
+  }
+
+  private HdfsConfiguration setUpHdfsConf() {
+    HdfsConfiguration conf = new HdfsConfiguration();
```

Comment:
why not inject hdfsconfiguration as well ?

Reply:
What do you mean by "inject"? Use Guice to inject it? Something else?

Reply:
yah, use Guice or somethign to @Inject it, like the one above

Reply:
I'll look into it.